src/Model/RecordLocale.php

```diff
@@ -315,4 +344,48 @@ public function IsDraft()
 
         return true;
     }
+
+    /**
+     * Check if a record is modified
+     * useful to indicate that record has changes that need publishing
+     *
+     * @return bool
+     */
+    public function IsModified()
+    {
+        if (!$this->IsPublished()) {
+            return false;
+        }
+
+        /** @var DataObject|FluentVersionedExtension $record */
+        $record = $this->getOriginalRecord();
+
+        return $record->stagesDifferInLocale($this->getLocale());
```

Comment:
If calling FluentVersionedExtension specific calls, please check for the extension on the record first.

Reply:
Good catch :)